### PR TITLE
feat: use flatten-maven-plugin to resolve bom imports

### LINKED ISSUE #254: Investigate maven-flatten-plugin
See if we can use maven-flatten-plugin so that the published google-cloud-bom has all artifact versions at the top level, rather than importing other boms.

We want to keep using imports of other boms for maintenance sake (updating one version at a time when other multi-module artifacts are published)

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #254

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -638,6 +638,26 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
```
Comment (saturnism) [inline pom.xml]: how's java-cloud-bom being used? typically plugins and common build functionalities do not go into a bom.

Comment (chingor13) [reply pom.xml]: This is being imported into `cloud.google.com:libraries-bom`. The build configuration is necessary to release this `pom.xml`. It's not (at least should not be) used as any project's parent.

Comment (saturnism) [reply pom.xml]: got it, then adding this plugin probably won't have any real/intended effects.

Comment (saturnism) [reply pom.xml]: let's sync up w/ stephanie and ben to discuss the details?
also, there is a current doc on the design of our build/pom relatoinships?

**Pull request closed without merging**